src/test/JwtUtils.test.js

```diff
@@ -0,0 +1,82 @@
+import {validateSchema} from "../utils/schemas/validateSchema";
+import {tokenPayloadSchema} from "../utils/schemas/tokenPayloadSchema";
+import {jwtUtils} from "../utils/jwt/jwtUtils";
+
+it('user token payload validates', () => {
+    const user = {
+        firstName: 'string',
+        lastName: 'string',
+        mail: 'string',
+        iss: '20',
+        iat: 10,
+        exp: 20
+    }
+    expect(validateSchema(user, tokenPayloadSchema)).toBe(true)
+})
+
+it('user token payload without field does not validate', () => {
+    const notUser = {
+        firstName: 'string',
+        lastName: 'string',
+        mail: 'string',
+        iat: 10
+    }
+    expect(validateSchema(notUser, tokenPayloadSchema)).toBe(false)
+})
+
+it('user token payload with incorrect field does not validate', () => {
+    const notUser = {
+        firstName: 'string',
+        lastName: 'string',
+        mail: 'string',
+        iss: '20p',
+        iat: 20,
+        exp: 10
+    }
+    expect(validateSchema(notUser, tokenPayloadSchema)).toBe(false)
+})
+
+it('token payload should be parsed properly', () => {
+    const token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiIxIiwiZmlyc3ROYW1lIjoiSm9obiIsImxhc3ROYW1lIjoiRG9lIiwibWFpbCI6Im1haWxAbWFpbC5jb20ifQ.pSfH9bWF6H1hJMFCgn35YhR0hgI8uFur9XcKKj9fze4"
```

Comment:
to jest jakiś ranodmowy token, czy legitny? Jeśli legitny, to nie wiem, czy dobrym pomysłem jest go trzymać w testach - w sensie wiem, że on się kiedyś zexpiruje, ale jednak still

Reply:
@Pinioo

Reply:
To tylko testy dla payloadu i poprawności funkcji do tych tokenów, sekret do hashowania jest jakiś z kosmosu, ten token nie jest nawet wygenerowany przez nasz serwis